### PR TITLE
[feat] resharding improvements

### PR DESCRIPTION
**Pull request by wacban**
- added some new metrics that are useful for resharding - shard layout version, shard layout num shards
- adjusted the existing resharding metrics to always report for the children shard uids
- added a few debug logs

### CODE REVIEW

File: chain/chain/src/chain.rs [hunk 1]
```diff
@@ -5512,6 +5515,10 @@ impl<'a> ChainUpdate<'a> {
                         .save_epoch_light_client_block(&prev_epoch_id.0, light_client_block);
                 }
             }
+
+            let shard_layout = self.epoch_manager.get_shard_layout_from_prev_block(prev.hash())?;
+            SHARD_LAYOUT_VERSION.set(shard_layout.version() as i64);
+            SHARD_LAYOUT_NUM_SHARDS.set(shard_layout.num_shards() as i64);
```
Comment (wacban) [inline chain/chain/src/chain.rs:5521]: When reviewing it's good to think about the placement of those lines. I placed it in postprocess_block, after the head is updated and in the metrics I report the values of the new head.

Comment (shreyan-gupta) [reply chain/chain/src/chain.rs:5521]: Should we have this emit only when `self.epoch_manager.will_shard_layout_change(parent_hash)` is true?

Comment (shreyan-gupta) [reply chain/chain/src/chain.rs:5521]: Why are we emitting this for every block processed?

**Review by shreyan-gupta — APPROVED** [review]

File: chain/chain/src/chain.rs (hunk 1 above)
Comment (wacban) [reply chain/chain/src/chain.rs:5521]: Yeah, I think so, it's only two numbers per block.

**Pull request merged**